ReactApp/src/sections/create-exam/add-question-to-exam-popup.js

```diff
@@ -0,0 +1,85 @@
+import { overlayStyle, popupStyle } from '../popUps/popup-style';
+import { Button, Typography, Divider, Box, FormControlLabel, Switch } from '@mui/material';
+import React, { useState } from 'react';
+import { EXAM } from '../../constants';
+import StemSelection from './stem-selection';
+import KeySelection from './keys-selection';
+import DistractorSelection from './distractor-selection';
+
+export const AddQuestionToExamPopup = ({isOpen, closePopup, metaQuestions, addQuestion}) => {
+
+  const [selectedMetaQuestion, setSelectedMetaQuestion] = useState(null);
+  const [selectedKey, setSelectedKey] = useState(null);
+  const [selectedDistractors, setSelectedDistractors] = useState([]);
+  const [generateState, setGenerateState] = useState(false); // Use boolean for simplicity
+
+  const handleSaveQuestion = () => {
+    addQuestion({
+      stem: selectedMetaQuestion.stem,
+      appendix: selectedMetaQuestion.appendix,
+      key: selectedKey,
+      distractors: selectedDistractors
+    }, generateState);
```

Comment:
Please pass the id as well.

I think a bettwe approch should be 

addQuestion{...selectedMetaQuestion, key:selectedKey, distractors:selectdDistractors, ect....}

Please just make sure to put the '...selectedMetaQuestion' at the beginning (due to js behaviour)